### PR TITLE
SN-20/User API

### PR DESCRIPTION
**Pull request by skripchenkoVeta**

### CODE REVIEW

File: app/controllers/api/v1/profiles_controller.rb [hunk 1]
```diff
@@ -51,21 +51,11 @@ def create
...
-        @profile = Profile.find(params[:id])
-      end
-
-      def serialize_profile
-        ProfileSerializer.new(@profile).to_json
-      end
-
-      def serialize_profiles
-        @profiles.map { |profile| ProfileSerializer.new(profile) }.to_json
+        @profile = Profile.find_by(user_id: params[:user_id]) || Profile.find(params[:id])
```
Comment (ramantechart) [inline app/controllers/api/v1/profiles_controller.rb]: we can get profile from  a user.
as soon as it's sunder resources user scope, we should define URLs like `users/:id/profile` and make just
```
Profile.find_by(user_id: ...)
```

File: config/routes.rb [hunk 2]
```diff
@@ -1,9 +1,16 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
-  mount_devise_token_auth_for "User", at: "auth"
+  mount_devise_token_auth_for 'User', at: 'auth'
   namespace :api, defaults: { format: :json } do
     namespace :v1 do
+      # mount_devise_token_auth_for 'User', at: 'auth'
       resources :profiles, only: %i[index show update create]
```
Comment (ramantechart) [inline config/routes.rb]: this should be under users resource

File: app/controllers/api/v1/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    # User Endpoints Controller
+    class UsersController < ApplicationController
+      before_action :authenticate_user!
```
Comment (ramantechart) [inline app/controllers/api/v1/users_controller.rb]: ditto

File: app/serializers/user_serializer.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+# class responsible for User Serialization
+class UserSerializer < ActiveModel::Serializer
```
Comment (ramantechart) [inline app/serializers/user_serializer.rb:4]: Keep in mind we have one more serializer for users.

**Review by ramantechart — CHANGES_REQUESTED** [review]

**Review by PivovarAndrew — APPROVED** [review]

**Pull request merged**